### PR TITLE
Return sc errors to both SQL clients and comdb2sc

### PR DESCRIPTION
**Pull request by riverszhang89**
There're a few cases where we do not return any errors to clients (e.g., index name too long). The patch fixes them, as well.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/362323dd/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>3/404 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>lostwrite [setup failure]<br>sc_timepart_logicalsc_generated<br>logfill</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/362323dd/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/362323dd/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2822'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/formatting-a-pr.html'>Use clang-format</a> or <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f4304bc5/format.diff'>view diff</a> &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/404 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>serialstep [setup failure]<br>op_makerecord</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f4304bc5/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/f4304bc5/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2822'>Resubmit</a></p>

File: cdb2api/cdb2api.h [hunk 1]
```diff
@@ -89,6 +89,7 @@ enum cdb2_errors {
     CDB2ERR_MALLOC = 115,
     CDB2ERR_NOTSUPPORTED = 116,
 
+    CDB2ERR_SCHEMACHANGE = 240,
```
Comment (nirbhayc) [inline cdb2api/cdb2api.h:92]: Can we reuse CDB2ERR_SCHEMA instead of adding a new code?

Comment (riverszhang89) [reply cdb2api/cdb2api.h:92]: `CDB2ERR_SCHEMA` indicates that the table version has changed.

**Review by nirbhayc — APPROVED** [review]
Looks great! Okay to merge.

**Comment by riverszhang89** [comment]
Thank you, Nirbhay!

**Pull request merged**